File: sunnad-ios/sunnad-ios/Domain/Repositories/CompletionsRepository.swift
import Foundation

protocol CompletionsRepository: Sendable {
    func fetchCompletions(on day: Date, calendar: Calendar, timeZone: TimeZone) async throws -> [HabitCompletion]
    func fetchCompletions(for habitID: UUID) async throws -> [HabitCompletion]
    func fetchCompletion(habitID: UUID, on day: Date, calendar: Calendar, timeZone: TimeZone) async throws -> HabitCompletion?
    func upsertCompletion(_ completion: HabitCompletion, calendar: Calendar, timeZone: TimeZone) async throws
    func deleteCompletion(habitID: UUID, on day: Date, calendar: Calendar, timeZone: TimeZone) async throws
}

File: sunnad-android/app/src/main/java/com/arystan/almasuly/sunnadandroid/domain/repository/CompletionsRepository.kt
package com.arystan.almasuly.sunnadandroid.domain.repository

import com.arystan.almasuly.sunnadandroid.core.model.HabitCompletion
import java.time.LocalDate
import java.util.UUID

interface CompletionsRepository {
    suspend fun fetchCompletions(day: LocalDate): List<HabitCompletion>
    suspend fun fetchCompletions(habitId: UUID): List<HabitCompletion>
    suspend fun fetchCompletion(habitId: UUID, day: LocalDate): HabitCompletion?
    suspend fun upsertCompletion(completion: HabitCompletion)
    suspend fun deleteCompletion(habitId: UUID, day: LocalDate)
}
